### ISSUE TITLE
Lars optimizer with AdaSum at hovorod, pytorch

### ISSUE THREAD

**Issue by xiaocaijizzz** [horovod/horovod#2914]
**Environment:**
1. Framework: PyTorch
2. Framework version: 
3. Horovod version: 
4. MPI version:
5. CUDA version:
6. NCCL version: 
7. Python version:
8. Spark / PySpark version:
9. Ray version:
10. OS and version:
11. GCC version:
12. CMake version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe erroneous behavior you're observing and steps to reproduce it.
Has anyone tried LARS optimizer at Horovod with AdaSum? I use HOROVOD_GPU_OPERATIONS=NCCL flag to compile Horovod. According to the official statement of Hovorod, only averaging through NCCL library is used to perform reductions and no AdaSum algorithm will take place when training is run on a single node. This means I will get the same result if the training is run on a single node, no matter which op I set, AdaSum or Average. And it is true when I use SGD optimizer. But when I use LARS optimizer, even if the training is run on a single node and the seed is fixed, I get quite different results when I set the distribution op as AdaSum and Average. I alse try Lamb, Adam, Adamw optimizers at hovorod in a single-node training, and the results are also different when the distribution op is set as Average or AdaSum. 
So I'd like to ask if anyone has verified AdaSum at Hovorod with Adam/Adamw/LARS/Lamb optimizer, especially in single-node training case. Has anyone tried Adam/Adamw/LARS/Lamb optimizer at hovorod and compare the weight updating of AdaSum with that of Average.

**Comment by Tixxx** [comment]
Hi @xiaocaijizzz thanks for reporting the issue. I have tried both Adam and Lamb on a single node with 4 gpus and the results between AdaSum and Average were extremely close for both fp32 and fp16. Just to confirm that you didn't specifically set any custom process group?
It would be great if you could share some repro code.
